diff -ru ./SD/src/utility/FatStructs.h ~/SD/src/utility/FatStructs.h
--- ./SD/src/utility/FatStructs.h	2018-11-22 08:48:46.000000000 -0500
+++ ~/SD/src/utility/FatStructs.h	2018-12-23 15:15:24.712996600 -0500
@@ -413,6 +413,6 @@
 }
 /** Directory entry is for a file or subdirectory */
 static inline uint8_t DIR_IS_FILE_OR_SUBDIR(const dir_t* dir) {
-  return (dir->attributes & DIR_ATT_VOLUME_ID) == 0;
+  return (dir->attributes & (DIR_ATT_VOLUME_ID | DIR_ATT_SYSTEM | DIR_ATT_HIDDEN)) == 0
 }
 #endif  // FatStructs_h
diff -ru ./SD/src/utility/SdFile.cpp ~/SD/src/utility/SdFile.cpp
--- ./SD/src/utility/SdFile.cpp	2018-11-22 08:48:46.000000000 -0500
+++ ~/SD/src/utility/SdFile.cpp	2018-12-23 15:15:59.920942800 -0500
@@ -268,7 +268,7 @@
       while ((b = *p++)) if (b == c) return false;
 #endif
       // check size and only allow ASCII printable characters
-      if (i > n || c < 0X21 || c > 0X7E)return false;
+      if (i > n || c < 0x20 || c == 0x7F) return false;
       // only upper case allowed in 8.3 names - convert lower to upper
       name[i++] = c < 'a' || c > 'z' ?  c : c + ('A' - 'a');
     }
