diff -ru ./SD/src/SD.cpp ~/SD/src/SD.cpp
--- ./SD/src/SD.cpp	2017-01-05 11:00:14.000000000 -0500
+++ ~/SD/src/SD.cpp	2018-05-15 11:11:41.191536900 -0500
@@ -342,7 +342,8 @@
     Return true if initialization succeeds, false otherwise.
 
    */
-  return card.init(SPI_HALF_SPEED, csPin) &&
+  if (root.isOpen()) root.close();
+  return card.init(SPI_FULL_SPEED, csPin) &&
          volume.init(card) &&
          root.openRoot(volume);
 }
diff -ru ./SD/src/utility/FatStructs.h ~/SD/src/utility/FatStructs.h
--- ./SD/src/utility/FatStructs.h	2017-01-05 11:00:14.000000000 -0500
+++ ~/SD/src/utility/FatStructs.h	2018-05-15 11:11:41.126732900 -0400
@@ -413,6 +413,6 @@
 }
 /** Directory entry is for a file or subdirectory */
 static inline uint8_t DIR_IS_FILE_OR_SUBDIR(const dir_t* dir) {
-  return (dir->attributes & DIR_ATT_VOLUME_ID) == 0;
+  return (dir->attributes & (DIR_ATT_VOLUME_ID | DIR_ATT_SYSTEM | DIR_ATT_HIDDEN)) == 0;
 }
 #endif  // FatStructs_h
diff -ru ./SD/src/utility/SdFile.cpp ~/SD/src/utility/SdFile.cpp
--- ./SD/src/utility/SdFile.cpp	2017-01-05 11:00:14.000000000 -0500
+++ ~/SD/src/utility/SdFile.cpp	2018-05-15 11:11:41.215595400 -0500
@@ -268,7 +268,7 @@
       while ((b = *p++)) if (b == c) return false;
 #endif
       // check size and only allow ASCII printable characters
-      if (i > n || c < 0X21 || c > 0X7E)return false;
+      if (i > n || c < 0x20 || c == 0x7F) return false;
       // only upper case allowed in 8.3 names - convert lower to upper
       name[i++] = c < 'a' || c > 'z' ?  c : c + ('A' - 'a');
     }
